src/main/java/tutorlink/lists/GradeList.java

```diff
@@ -53,12 +54,27 @@ public GradeList findGrade(String matricNumber, String componentDescription) thr
         filteredList.gradeArrayList = gradeArrayList
                 .stream()
                 .filter(grade -> grade.getStudent().getMatricNumber().equals(matricNumber.toUpperCase())
-                && grade.getComponent().getName().equals(componentDescription.toLowerCase()))
+                        && grade.getComponent().getName().equals(componentDescription.toLowerCase()))
                 .collect(Collectors.toCollection(ArrayList::new));
         if (filteredList.gradeArrayList.isEmpty()) {
             throw new GradeNotFoundException(String.format(ERROR_NO_GRADE_FOUND, matricNumber, componentDescription));
         }
         return filteredList;
     }
 
+    public double calculateStudentGPA(String matricNumber) {
+        ArrayList<Grade> studentGrades = gradeArrayList
+                .stream()
+                .filter(grade -> grade.getStudent().getMatricNumber().equals(matricNumber.toUpperCase()))
+                .collect(Collectors.toCollection(ArrayList::new));
+
+        if (studentGrades.isEmpty()) {
+            return 0.0;
+        }
+
+        return studentGrades
+                .stream()
+                .mapToDouble(grade -> grade.getScore() * grade.getComponent().getWeight())
+                .sum();
+    }
```

Comment:
Just to confirm that score is a fraction that represents the percentage score. 

What happens if not all the grades are in yet? Currently it seems the GPA will still be computed assuming that the pending grades are 0. Not sure if this is intended behaviour. 

Otherwise looks good to me.

Reply:
It looks like it has been rectified, LGTM